core/blockchain.go

```diff
@@ -3015,3 +3034,155 @@ func EnablePersistDiff(limit uint64) BlockChainOption {
 		return chain
 	}
 }
+
+func (bc *BlockChain) GetRootByDiffHash(blockNumber uint64, blockHash common.Hash, diffHash common.Hash) (*types.VerifyResult, error) {
+	var res types.VerifyResult
+	res.BlockNumber = blockNumber
+	res.BlockHash = blockHash
+
+	if blockNumber > bc.CurrentHeader().Number.Uint64()+11 {
```

Comment:
`11` is a magic number here, lets name it.

Reply:
sure, replace by maxDiffForkDist

Reply:
Maybe `maxForkDist` is better, so other place can use it too.

Reply:
maxDiffForkDist has been defined in the core package, it seems use maxDiffForkDist is ok.